### PR TITLE
vendor: bump hcsshim to v0.8.22

### PR DESCRIPTION
**Pull request by dcantah**
#### What type of PR is this?
/kind bug
/kind cleanup

#### What this PR does / why we need it:
This tag of hcsshim brings in a couple welcome features/improvements. One being exposing a way to query for hns endpoint statistics (Packets received/sent etc.). This tag also contains some optimizations for querying whether a certain HCN feature
is supported, which is a common workflow in kube-proxy on Windows. The first result from querying HCN is now cached so further calls can skip the hcn query as well as the version range parsing that was performed. This also gets rid of some redundant logs that used to hit everytime the version range parsing occurred.

The Go-winio dep bump, and all of the ctrd deps are transitive only.Nnothing new is needed/intended
to be used.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dcantah** [comment]
Testing out a local branch to see if the tests pass first, then will swap to the actual tag.

**Comment by marosset** [comment]
/ok-to-test
(so we can get a signal before making a new hcsshim release)

File: go.mod [hunk 1]
```diff
@@ -139,6 +139,7 @@ require (
 )
 
 replace (
+	bazil.org/fuse => bazil.org/fuse v0.0.0-20160811212531-371fbbdaa898
```
Comment (liggitt) [inline go.mod:142]: I'm surprised this didn't show up as a new dep in depstats... it's not in our dependency tree currently, right? what is pulling this in?

Comment (dcantah) [reply go.mod:142]: That's what I was wondering too after just noticing... No I don't believe so, I'll go on a hunt in a second.

Comment (liggitt) [reply go.mod:142]: answering my own question, github.com/containerd/continuity pulls it in, and did even on the version we were on before (https://github.com/containerd/continuity/blob/aaeac12a7ffcd198ae25440a9dff125c2e2703a7/continuityfs/fuse.go) but did not have a go.mod file in that version so it didn't show up in the graph

Comment (dcantah) [reply go.mod:142]: Welp, checked back in to answer and it's already checked in 😂. Glad you found it

**Pull request merged**